src/AutorestSwift/View Models/OperationViewModel.swift

```diff
@@ -35,6 +35,10 @@ struct OperationViewModel {
     let comment: ViewModelComment
     let params: [ParameterViewModel]
     let returnType: ReturnTypeViewModel?
+    // Query Params/Header can be placed inside QueryParameter Initializer
+    let queryInitParams: [KeyValueViewModel]?
+    let headerInitParams: [KeyValueViewModel]?
+    // Query Params/Header need to add Nil check
```

Comment:
I don't think you need to separate these out. It is totally fine to instantiate an empty dictionary for queryParams and headers and then simply add each with subsequent lines (i.e. don't bother to populate in the initializer).